src/QsCompiler/Tests.Compiler/TypeCheckingTests.fs

```diff
@@ -62,14 +58,19 @@ module TypeCheckingTests =
 
     [<Fact>]
     let ``Supports sized array literals`` () =
-        expect "SizedArray1" []
-        expect "SizedArray2" []
-        expect "SizedArray3" []
-        expect "SizedArray4" []
+        allValid "SizedArray" 4
         expect "SizedArrayInvalid1" [ Error ErrorCode.TypeMismatch ]
         expect "SizedArrayInvalid2" [ Error ErrorCode.TypeMismatch ]
         expect "SizedArrayInvalid3" [ Error ErrorCode.TypeMismatch ]
 
+    [<Fact>]
+    let ``Supports lambda expressions`` () =
+        allValid "Lambda" 14
+        expect "LambdaInvalid1" [ Error ErrorCode.TypeMismatchInReturn ]
+        expect "LambdaInvalid2" [ Error ErrorCode.TypeMismatchInReturn ]
+        expect "LambdaInvalid3" (Error ErrorCode.InfiniteType |> List.replicate 2)
+// TODO: expect "LambdaInvalid4" [ (* Closing over mutable *) ]
```

Comment:
FYI: Recently we were discussing with Troels and Bettina various coding styles. And one item attracted my attention in one of the styles. The item stated that every "TODO" must be followed by an owner alias or a bug number. I liked that item and adopted for myself. Now I write like this: `TODO(rokuzmin)`.

Reply:
I updated the todo comment to include a link to an issue.